### PR TITLE
Playback (direct-stream/transcode) of BDISO/BDAV containers

### PR DESCRIPTION
**Pull request by stanionascu**
As some people prefer storing the backups of the original media in an untouched containers, it would be great if Jellyfin would support playback (remux/transcoding) from them.

**Notes**
ffmpeg/ffprobe built with libbluray, supports direct remuxing/transcoding of the main movie (including BDJ).

With this commit the media info of the ISO can be identified and playback is working without relying on BDInfo.

**Changes**
* Support playback from BD and UHD ISO containers.
* Support playback from BDAV / UHD-BDAV folders.

**Issues**
N/A

### CODE REVIEW

**Comment by stanionascu** [comment]
Screnshot:
screen

**Comment by ferferga** [comment]
Thank you very much for your work. However, this should be added in a plugin imo.

What do you think @jellyfin/server ?

**Comment by stanionascu** [comment]
> Thank you very much for your work. However, this should be added in a plugin imo.
> 
> What do you think @jellyfin/server ?

That would be ideal, but I don't see how a plugin can provide special paths that are passed into ffprobe/ffmpeg. If the MediaProtocol is File, then the InputFileArgument with start with `file://`. To support playback and media-identification, it is necessary to use the paths as `bluray:/path/to/iso` or `bluray:/path/to/bdav`.  

This PR already allows playback from ISO/BDAV containers, without relying on BDInfo. But I'm personally not too happy with the code changes.

**Comment by EraYaN** [comment]
> 
> 
> Thank you very much for your work. However, this should be added in a plugin imo.
> 
> What do you think @jellyfin/server ?

I think this can and probably should be in core. ISOs are common enough and ffmpeg supports them quite well these days.

**Comment by crobibero** [comment]
is this PR still a WIP, or is it ready for review?

**Comment by stanionascu** [comment]
> is this PR still a WIP, or is it ready for review?

Since there are no more comments/suggestions, it should be ready for review now.

**Review by crobibero — APPROVED** [review]

**Comment by crobibero** [comment]
Looks like 99% cleanup, with a file protocol change for support.
After this PR `IIsoManager` and `IsoManager` can be removed as they are no longer used.

**Comment by stanionascu** [comment]
> Looks like 99% cleanup, with a file protocol change for support.
> After this PR `IIsoManager` and `IsoManager` can be removed as they are no longer used.

I can also remove them with this PR.

> 
> --
> It looks like libbluray is enabled in `jellyfin-ffmpeg`, is this version of ffmpeg what you tested with?

I tested with latest stable `scoop` installed ffmpeg from the environment. I'll re-test with jellyfin-ffmpeg.

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -2368,18 +2352,10 @@ public void TryStreamCopy(EncodingJobInfo state)
             }
         }
 
-        public string GetProbeSizeArgument(int numInputFiles)
```
Comment (nyanmisaka) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2371]: Why remove these `FFmpeg:probesize` and `FFmpeg:analyzeduration`  environment variables? 
Relates to https://github.com/jellyfin/jellyfin/issues/3930

**Review by nyanmisaka — CHANGES_REQUESTED** [review]
These variables are in use in some cases.

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs (hunk 1 above)
Comment (stanionascu) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs:2371]: After this change the `numInputFiles` becomes always 1, thus ther result of the method will be always `string.Empty`

**Review by nyanmisaka — APPROVED** [review]
LGTM

**Pull request merged**